users/views.py

```diff
@@ -0,0 +1,56 @@
+from django.contrib.auth.tokens import default_token_generator
+from rest_framework import status, viewsets
+from rest_framework.decorators import action, api_view
+from rest_framework.response import Response
+from rest_framework.reverse import reverse
+
+from users.models import User
+from users.serializers import UserSerializer
+
+
+@api_view(['GET'])
+def api_root(request, format=None):
+    return Response({
+        'users': reverse('users:user_list', request=request, format=format),
+    })
+
+
+class UserViewSet(viewsets.ModelViewSet):
```

Comment:
Can we use other types of views from rest_framework which can fit better in this place ?

Reply:
I could use ViewSet and implement list and create actions or create custom ViewSet inheriting from GenericViewSet and List and Create Mixins.
My idea was to implement ModelViewSet, but include in routes only needed actions (otherwise I could have used DRF's default router instead of mapping the actions in urls). I thought it would make easier to extend user's functionalities if needed (only add new route). As I understand, if the action is not mapped to an url, it wont' be accessible and used.
I'm not saying it'a a good approach, I just wanted to try this one ;)